samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * @file JobsAgent.cpp
+ *
+ * This is a full featured sample agent that interacts with the AWS IoT Jobs 
+ * platform. For more information see the README.md file in the samples directory.
+ *
+ */
+
+#include <chrono>
+#include <cstring>
+#include <sys/utsname.h>
+
+#ifdef USE_WEBSOCKETS
+#include "WebSocketConnection.hpp"
+#elif defined USE_MBEDTLS
+#include "MbedTLSConnection.hpp"
+#else
+#include "OpenSSLConnection.hpp"
+#endif
+
+#include "util/logging/Logging.hpp"
+#include "util/logging/LogMacros.hpp"
+#include "util/logging/ConsoleLogSystem.hpp"
+
+#include "ConfigCommon.hpp"
+#include "jobs/Jobs.hpp"
+#include "JobsAgent.hpp"
+
+namespace awsiotsdk {
+    namespace samples {
+        ResponseCode JobsAgent::InitializeTLS() {
+            ResponseCode rc = ResponseCode::SUCCESS;
+
+#ifdef USE_WEBSOCKETS
+            p_network_connection_ = std::shared_ptr<NetworkConnection>(
+                new network::WebSocketConnection(ConfigCommon::endpoint_, ConfigCommon::endpoint_https_port_,
+                                                 ConfigCommon::root_ca_path_, ConfigCommon::aws_region_,
+                                                 ConfigCommon::aws_access_key_id_,
+                                                 ConfigCommon::aws_secret_access_key_,
+                                                 ConfigCommon::aws_session_token_,
+                                                 ConfigCommon::tls_handshake_timeout_,
+                                                 ConfigCommon::tls_read_timeout_,
+                                                 ConfigCommon::tls_write_timeout_, true));
+            if (nullptr == p_network_connection_) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "Failed to initialize Network Connection. %s",
+                              ResponseHelper::ToString(rc).c_str());
+                rc = ResponseCode::FAILURE;
+            }
+#elif defined USE_MBEDTLS
+            p_network_connection_ = std::make_shared<network::MbedTLSConnection>(ConfigCommon::endpoint_,
```

Comment:
hmmm.... so make_shared here but not above? Is there a reason or is it just an oversight?